x/bsonx/bsoncore/array_test.go

```diff
@@ -348,236 +349,77 @@ func TestArray(t *testing.T) {
-				{
-					Type: TypeString,
-					Data: AppendString(nil, "abc"),
-				},
-				{
-					Type: TypeInt32,
-					Data: AppendInt32(nil, 123),
-				},
-				{
-					Type: TypeBoolean,
-					Data: AppendBoolean(nil, true),
-				},
-			},
-			want: "",
-		},
-
-		// n > 0 cases
-		{
-			description: "n>0, array LT n",
-			n:           1,
-			values: []Value{
-				{
-					Type: TypeInt32,
-					Data: AppendInt32(nil, 2),
-				},
-			},
-			want: "[",
-		},
-		{
-			description: "n>0, array LT n",
-			n:           2,
-			values: []Value{
-				{
-					Type: TypeInt32,
-					Data: AppendInt32(nil, 2),
-				},
-			},
-			want: "[{",
-		},
-		{
-			description: "n>0, array LT n",
-			n:           14,
-			values: []Value{
-				{
-					Type: TypeInt32,
-					Data: AppendInt32(nil, 2),
-				},
-			},
-			want: `[{"$numberInt"`,
-		},
-		{
-			description: "n>0, array GT n",
-			n:           30,
-			values: []Value{
-				{
-					Type: TypeInt32,
-					Data: AppendInt32(nil, 2),
-				},
-			},
-			want: `[{"$numberInt":"2"}]`,
-		},
-		{
-			description: "n>0, array EQ n",
-			n:           22,
-			values: []Value{
-				{
-					Type: TypeInt32,
-					Data: AppendInt32(nil, 2),
-				},
-			},
-			want: `[{"$numberInt":"2"}]`,
-		},
-		{
-			description: "n>0, mixed array",
-			n:           24,
-			values: []Value{
-				{
-					Type: TypeInt32,
-					Data: AppendInt32(nil, 1),
-				},
-				{
-					Type: TypeString,
-					Data: AppendString(nil, "foo"),
-				},
-			},
-			want: `[{"$numberInt":"1"},"foo`,
-		},
-		{
-			description: "n>0, empty array",
-			n:           10,
-			values:      []Value{},
-			want:        "[]",
-		},
-		{
-			description: "n>0, nested array",
-			n:           10,
-			values: []Value{
-				{
-					Type: TypeArray,
-					Data: BuildArray(nil, Value{
-						Type: TypeString,
-						Data: AppendString(nil, "abc"),
-					}),
-				},
-			},
+				}),
+			}),
 			want: `[["abc"]]`,
 		},
 		{
-			description: "n>0, array with mixed types",
-			n:           32,
-			values: []Value{
-				{
+			description: "array with mixed types",
+			array: BuildArray(nil,
+				Value{
 					Type: TypeString,
 					Data: AppendString(nil, "abc"),
 				},
-				{
+				Value{
 					Type: TypeInt32,
 					Data: AppendInt32(nil, 123),
 				},
-				{
+				Value{
 					Type: TypeBoolean,
 					Data: AppendBoolean(nil, true),
 				},
-			},
-			want: `["abc",{"$numberInt":"123"},true`,
+			),
+			want: `["abc",{"$numberInt":"123"},true]`,
 		},
 	}
 
 	for _, tc := range testCases {
-		t.Run(tc.description, func(t *testing.T) {
-			got := Array(BuildArray(nil, tc.values...)).StringN(tc.n)
+		t.Run(fmt.Sprintf("String %s", tc.description), func(t *testing.T) {
+			got := tc.array.String()
 			assert.Equal(t, tc.want, got)
-			if tc.n >= 0 {
-				assert.LessOrEqual(t, len(got), tc.n)
-			}
 		})
 	}
+
+	for _, tc := range testCases {
+		for n := -1; n <= len(tc.want)+1; n++ {
+			t.Run(fmt.Sprintf("StringN %s n==%d", tc.description, n), func(t *testing.T) {
```

Comment:
Optional: Consider factoring these into two test functions `TestArray_String` and `TestArray_StringN` and moving the test cases to a package-level variable (e.g. `var arrayStringTestCases = ...`).